commit af4c566f1bdeeb385828a8fbd4b1248a6f6acc04
Author: Arne Ruhnau <arne.ruhnau@target-sg.com>
Date:   Wed Jul 16 23:46:53 2014 +0200

    Initial support for the WLAN module.
    
    The module does not work after soft-reset via reboot.

diff --git a/arch/arm/boot/dts/imx6dl-trizeps7-pconxs.dts b/arch/arm/boot/dts/imx6dl-trizeps7-pconxs.dts
index 3e87a25..cf33dbd 100644
--- a/arch/arm/boot/dts/imx6dl-trizeps7-pconxs.dts
+++ b/arch/arm/boot/dts/imx6dl-trizeps7-pconxs.dts
@@ -312,12 +312,16 @@
 	status = "okay";
 };
 
-/*
-&usdhc1 {
+&usdhc1 { /* this is the WiBear 11n wireless module */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc1_2>;
 	status = "okay";
-	vmmc-supply = <&sw4_reg>;
+/* supply comes from vgen4 and vgen3 ... */
+/* there's reset (&gpio1 2 0) and power down (&gpio4 10 0) GPIOs */
+	non-removable;
+	keep-power-in-suspend;
+	no-1-8-v;
+	bus-width = <4>;
 };
 */
 &usdhc2 {
diff --git a/firmware/LICENCE.Marvell b/firmware/LICENCE.Marvell
new file mode 100644
index 0000000..3224e1b
--- /dev/null
+++ b/firmware/LICENCE.Marvell
@@ -0,0 +1,18 @@
+FIRMWARE LICENSE TERMS
+
+
+Copyright (c) Marvell International Ltd.
+
+All rights reserved.
+
+Redistribution. Redistribution and use in binary form, without modification, are permitted provided that the following conditions are met:
+
+* Redistributions must reproduce the above copyright notice and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+* Neither the name of Marvell International Ltd. nor the names of its suppliers may be used to endorse or promote products derived from this software without specific prior written permission.
+
+* No reverse engineering, decompilation, or disassembly of this software is permitted.
+
+Limited patent license. Marvell International Ltd. grants a world-wide, royalty-free, non-exclusive license under patents it now or hereafter owns or controls to make, have made, use, import, offer to sell and sell ("Utilize") this software, but solely to the extent that any such patent is necessary to Utilize the software alone, or in combination with an operating system licensed under an approved Open Source license as listed by the Open Source Initiative at http://opensource.org/licenses. The patent license shall not apply to any other combinations which include this software. No hardware per se is licensed hereunder.
+
+DISCLAIMER. THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
diff --git a/firmware/Makefile b/firmware/Makefile
index 4615b03..ec4c76b 100644
--- a/firmware/Makefile
+++ b/firmware/Makefile
@@ -137,6 +137,7 @@ fw-shipped-$(CONFIG_VIDEO_CPIA2) += cpia2/stv0672_vp4.bin
 fw-shipped-$(CONFIG_YAM) += yam/1200.bin yam/9600.bin
 
 fw-shipped-${CONFIG_IMX_SDMA} += imx/sdma/sdma-imx6q.bin
+fw-shipped-${CONFIG_MWIFIEX_SDIO} += mrvl/sd8787_uapsta.bin
 
 fw-shipped-all := $(fw-shipped-y) $(fw-shipped-m) $(fw-shipped-)
 
diff --git a/firmware/WHENCE b/firmware/WHENCE
index c511c96..88f3b28 100644
--- a/firmware/WHENCE
+++ b/firmware/WHENCE
@@ -853,6 +853,16 @@ Found in hex form in kernel source.
 
 --------------------------------------------------------------------------
 
+Driver: mwifiex - Marvell Wi-Fi fullmac-type 802.11n/ac cards
+
+File: mrvl/sd8787_uapsta.bin
+Version: 14.66.9.p96
+
+Licence: Redistributable. See LICENCE.Marvell for details.  Originates from
+http://git.marvell.com/?p=mwifiex-firmware.git
+
+--------------------------------------------------------------------------
+
 Driver: imx-sdma - SDMA scripts
 
 File: imx/sdma/sdma-imx6q.bin.ihex
diff --git a/firmware/mrvl/sd8787_uapsta.bin b/firmware/mrvl/sd8787_uapsta.bin
new file mode 100644
index 0000000..3bf4e23
Binary files /dev/null and b/firmware/mrvl/sd8787_uapsta.bin differ
